### PR TITLE
feat: Optionally hide deprecation errors

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Allows users to hide the deprecation warnings and hides them while our internal tests are run.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch.
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [X] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Review by afontcu — APPROVED** [review]

**Comment by fredicious** [comment]
> Allows users to hide the deprecation warnings

What would be the way to do that ? the config `showDeprecationWarnings` doesn't seem to be supported yet

**Comment by lmiller1990** [comment]
@fredicious it should work - did you do this:

```
import { config } from '@vue/test-utils'
config.showDeprecationWarnings = false
```

In your test setup file? It should disable for all the tests.

**Comment by fredicious** [comment]
that worked, thanks a lot @lmiller1990 
Somehow I got confused between the main config and the mounting options.

File: packages/shared/util.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // @flow
 import Vue from 'vue'
 import semver from 'semver'
+import { config } from '@vue/test-utils'
```
Comment (sullivanpt) [inline packages/shared/util.js:4]: Not sure why, but this new import here breaks the IIFE. Would love some help tracking this down and fixing it. See #1553

**Pull request merged**